R/set-defaults.R

```diff
@@ -0,0 +1,166 @@
+#' # points but we can use this as a short cut to see some of them:
+#' names(compare_values(2, 3)) %>% head()
+#'
+#' # update headline
+#' set_headliner_defaults(
+#'   headline = "{article_delta_p}% {trend} ({orig_values})",
+#'   orig_values = "from {r} to {c}"
+#' )
+#'
+#' # view changes
+#' show_headliner_defaults()
+#'
+#' # and in action
+#' headline(3, 4)
+#'
+#' # reset values
+#' set_headliner_defaults(headline = NULL, orig_values = NULL)
+#' headline(3, 4)
+set_headliner_defaults <- function(headline = NULL,
+                                   orig_values = NULL) {
+  if (!missing(headline)) {
+    update_default(x = "headline", value = headline)
+  }
+
+  if (!missing(orig_values)) {
+    update_default(x = "orig_values", value = orig_values)
+  }
+}
+
+
+#' Update the values headliner uses in the active session
+#' @noRd
+update_default <- function(x, value) {
+  old <- headliner_global[[x]]
+  default <- get(paste0("default_", x))
+  if (is.null(value)) {
+    headliner_global[[x]] <- default
+  } else if (length(x) > 1) {
+    stop(
+      "'", x, "' should be length of 1 not ", length(get(x)),
+      call. = FALSE
+    )
+  } else {
+    headliner_global[[x]] <- value
+  }
+}
+
+
+
+
+
+# ARTICLES ----
+#' Adjust which words start with "a" vs "an"
+#'
+#' headliner uses crude logic to anticipate the articles used before phrases
+#' like "(an) increase", "(a) decrease", "(an) 83" and "(a) -5"
+#' You will likely not call the function [get_article()] directly but you
+#' can augment the logic it uses with this function.
+#'
+#' These arguments assume that the words start with the patterns you provide.
+#' Passing "hour|heir" will be translated to "^(hour|heir)".
+#' @param regex_for_a regular expression for words that get the article 'a'
+#' @param regex_for_an regular expression for words that get the article 'an'
+#' @export
+#' @seealso [show_headliner_defaults()], [set_headliner_defaults()]
+#' @examples
+#' # The crude logic for headliner would give all "h" words the article "a"
+#' # by default, the word "heirloom" returns "a"
+#' get_article("heirloom")
+#'
+#' # the patterns that are used can be updated
+#' augment_article_patterns(
+#'   regex_for_a = "Euro|uni",
+#'   regex_for_an = "hour|heir"
+#' )
+#'
+#' get_article("heirloom")
+#'
+#' # these patterns can be reset using 'NULL'
+#' augment_article_patterns(
+#'   regex_for_an = NULL
+#' )
+#'
+#' get_article("heirloom")
+augment_article_patterns <- function(regex_for_a, regex_for_an) {
+  old <- headliner_global$articles
+
+  if (!missing(regex_for_a)) {
+    a_patterns <- validate_augment_articles(regex_for_a)
+    headliner_global$articles$addl_a <- a_patterns
+  }
+
+  if (!missing(regex_for_an)) {
+    an_patterns <- validate_augment_articles(regex_for_an)
+    headliner_global$articles$addl_an <- an_patterns
+  }
+}
+
+
+validate_augment_articles <- function(x = NULL) {
+  if (length(x) > 1) {
+    stop(
+      "patterns should be a single regular expression",
```

Comment:
Is there a reason they couldn't pass a vector of regexs, and then you `collapse` them for the user?

Reply:
I considered this but didn't know if it'd be as intuitive and it makes it more verbose since the patterns are always "starts with". I can't imagine this function being called often. It (currently) only gets used for numbers and words that describe increases/decreases. Can you give an example where a vector would be perfered?

Reply:
Ah I see. I was just thinking about saving the user a step if they had a list of regexs they wanted to pass in --  but if it's rare then this works. I could also see this being easier than passing a vector, since you don't have to comma separate everything for a short list.